public/kibi_timeline.js

```diff
@@ -203,68 +203,87 @@ define(function (require) {
+                  notify.warning('Check your data - the number of values in the \'Event end date\'' +
+                  ' must be equal to the number of values in the \'Event start date\'');
+                  return;
                 }
+              }
+
+              if (startFieldValue && (!_.isArray(startFieldValue) || startFieldValue.length)) {
                 let indexId = searchSource.get('index').id;
-                let startValue = timelineHelper.pickFirstIfMultivalued(startFieldValue);
-                let startRawValue = timelineHelper.pickFirstIfMultivalued(startRawFieldValue);
-                let content =
-                  '<div title="index: ' + indexId +
-                  ', startField: ' + params.startField +
-                  (params.endField ? ', endField: ' + params.endField : '') +
-                  '">' + labelValue +
-                  (params.useHighlight ? '<p class="tiny-txt">' + timelineHelper.pluckHighlights(hit, highlightTags) + '</p>' : '') +
-                  '</div>';
-
-                let e =  {
-                  index: indexId,
-                  content: content,
-                  value: labelValue,
-                  start: new Date(startRawValue),
-                  startField: {
-                    name: params.startField,
-                    value: startValue
-                  },
-                  type: 'box',
-                  group: $scope.groupsOnSeparateLevels === true ? index : 0,
-                  style: 'background-color: ' + groupColor + '; color: #fff;',
-                  groupId: groupId
-                };
-
-                if (params.endField) {
-                  if (params.endFieldSequence) { // in kibi, we have the path property of a field
-                    endFieldValue = kibiUtils.getValuesAtPath(hit._source, params.endFieldSequence);
+
+                startFieldValue.forEach(function (value, i) {
+                  let startValue = value;
+                  let startRawValue = startRawFieldValue[i];
+
+                  var content;
+                  if (labelValue.length === startFieldValue.length) {
+                    content =
+                      '<div title="index: ' + indexId +
+                      ', startField: ' + params.startField +
+                      (params.endField ? ', endField: ' + params.endField : '') +
+                      '">' + labelValue[i] +
+                      (params.useHighlight ? '<p class="tiny-txt">' + timelineHelper.pluckHighlights(hit, highlightTags) + '</p>' : '') +
+                      '</div>';
                   } else {
-                    endFieldValue = _.get(hit._source, params.endField);
+                    content =
+                      '<div title="index: ' + indexId +
+                      ', startField: ' + params.startField +
+                      (params.endField ? ', endField: ' + params.endField : '') +
+                      '">' + labelValue.join() +
```

Comment:
Not sure but if we want the labels to be comma separated 

labelValue.join() -> labelValue.join(',')